engine/access/state_stream/backend/handler.go

```diff
@@ -215,3 +217,260 @@ func (h *Handler) GetRegisterValues(_ context.Context, request *executiondata.Ge
 
 	return &executiondata.GetRegisterValuesResponse{Values: values}, nil
 }
+
+func (h *Handler) createAccountStatusesFilter(statusFilter *executiondata.StatusFilter) (state_stream.EventFilter, error) {
+	filter := state_stream.EventFilter{}
+	if statusFilter != nil {
+		var err error
+		filter, err = state_stream.NewEventFilter(
+			h.eventFilterConfig,
+			h.chain,
+			statusFilter.GetEventType(),
+			[]string{},
+			[]string{},
+		)
+		if err != nil {
+			return filter, status.Errorf(codes.InvalidArgument, "invalid event filter: %v", err)
+		}
+
+		addresses := statusFilter.GetAddress()
+		if len(addresses) > 0 {
+			var fieldFilters []state_stream.FieldFilter
+			for _, address := range addresses {
+				fieldFilters = append(fieldFilters, state_stream.FieldFilter{
+					FieldName:   "address",
+					TargetValue: address,
+				})
+			}
+
+			for eventType := range filter.EventTypes {
+				filter.EventFieldFilters[eventType] = fieldFilters
+			}
+		}
+	}
+
+	return filter, nil
+}
+
+func convertAccountsStatusesResults(eventVersion entities.EventEncodingVersion, resp *AccountStatusesResponse) ([]*executiondata.SubscribeAccountStatusesResponse_Result, error) {
+	var results []*executiondata.SubscribeAccountStatusesResponse_Result
+	for address, events := range resp.AccountEvents {
+		convertedEvent, err := convert.EventsToMessagesWithEncodingConversion(events, entities.EventEncodingVersion_CCF_V0, eventVersion)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "could not convert events to entity: %v", err)
+		}
+
+		results = append(results, &executiondata.SubscribeAccountStatusesResponse_Result{
+			Address: []byte(address),
+			Events:  convertedEvent,
+		})
+	}
+	return results, nil
+}
+
+func (h *Handler) SubscribeAccountStatusesFromStartBlockID(request *executiondata.SubscribeAccountStatusesFromStartBlockIDRequest, stream executiondata.ExecutionDataAPI_SubscribeAccountStatusesFromStartBlockIDServer) error {
```

Comment:
any way to reduce the boilerplate here since the methods all return the same data type? Before making any changes, it's worth checking out what Uliana did in the block streaming PR to see if any patterns/code could be reused here.